miniapps/navier/navier_bifurcation.cpp

```diff
@@ -0,0 +1,362 @@
+   attr[1] = 1;
+   flow_solver.AddVelDirichletBC(vel_dbc, attr);
+
+#ifdef MFEM_USE_GSLIB
+   ParGridFunction &w_gf = *flow_solver.GetCurrentVorticity();
+   // Create the particle solver
+   NavierParticles particle_solver(MPI_COMM_WORLD, 0, pmesh);
+   int nparticles = ((ctx.nt - ctx.pnt_0)/ctx.add_particles_freq) *
+                    ctx.num_add_particles / Mpi::WorldSize();
+   particle_solver.GetParticles().Reserve(nparticles);
+
+   // Set particle BCs - left normal for line connecting start to end must
+   // point into the domain. If not, we set invert_normal to true.
+   particle_solver.Add2DReflectionBC(Vector({0.0, 1.0}), Vector({8.0, 1.0}),
+                                     1.0, true);
+   particle_solver.Add2DReflectionBC(Vector({8.0, 1.0}), Vector({8.0, 9.0}),
+                                     1.0, true);
+   particle_solver.Add2DReflectionBC(Vector({9.0, 9.0}), Vector({9.0, 1.0}),
+                                     1.0, true);
+   particle_solver.Add2DReflectionBC(Vector({9.0, 1.0}), Vector({17.0, 1.0}),
+                                     1.0, true);
+   particle_solver.Add2DReflectionBC(Vector({0.0, 0.0}), Vector({17.0, 0.0}),
+                                     1.0, false);
+#endif
+
+   // Set up solution visualization
+   char vishost[] = "localhost";
+   socketstream vis_sol;
+   int Ww = 350, Wh = 350; // window size
+   int Wx = 10, Wy = 0; // window position
+   char keys[] = "mAcRjlmm]]]]]]]]]";
+   if (ctx.visualization)
+   {
+      VisualizeField(vis_sol, vishost, ctx.visport, u_gf, "Velocity",
+                     Wx, Wy, Ww, Wh, keys);
+   }
+
+   // Initialize ParaView DC (if freq != 0)
+   std::unique_ptr<ParaViewDataCollection> pvdc;
+   if (ctx.paraview_freq > 0)
+   {
+      pvdc = std::make_unique<ParaViewDataCollection>("Bifurcation", &pmesh);
+      pvdc->SetPrefixPath("ParaView");
+      pvdc->SetLevelsOfDetail(ctx.order);
+      pvdc->SetDataFormat(VTKFormat::BINARY);
+      pvdc->SetHighOrderOutput(true);
+      pvdc->RegisterField("Velocity",flow_solver.GetCurrentVelocity());
+      pvdc->RegisterField("Vorticity",flow_solver.GetCurrentVorticity());
+      pvdc->SetTime(time);
+      pvdc->SetCycle(0);
+      pvdc->Save();
+   }
+
+#ifdef MFEM_USE_GSLIB
+   std::string csv_prefix = "Navier_Bifurcation_";
+   // Setup arrays to indicate what to print.
+   // Leave field array empty to only print minimal info, i.e. id, rank, and
+   // coordinates
+   Array<int> print_field_idxs;
+   // Print the first tag for each particle.
+   Array<int> print_tag_idxs(1);
+   print_tag_idxs[0] = 0;
+   if (ctx.print_csv_freq > 0)
+   {
+      std::string file_name = csv_prefix +
+                              mfem::to_padded_string(0, 6) + ".csv";
+      particle_solver.GetParticles().PrintCSV(file_name.c_str(),
+                                              print_field_idxs,
+                                              print_tag_idxs);
+   }
+   particle_solver.Setup(ctx.dt);
```

Comment:
Why not to visualize the particles with the new `ParticleTrajectories`? 👍  It is not used anywhere atm.

Reply:
It will be used in Lorentz. Maybe we should remove from this PR for now or just let it be and we can update in that PR if necessary?

Reply:
I have added GLVis visualization now. But I am afraid that since GLVis is not yet well suited for these kind of particle visualizations (e.g., trajectories are not really visible for small segment lengths and/or particle count), it might not be a good user experience. 

I see these new visualization methods as helper functions for quick prototyping and testing. I have added a note to the documentation of `ParticleTrajectories` that folks can switch to Paraview for more sophisticated use cases.